File: src/main/java/pw/switchcraft/plethora/integration/vanilla/meta/block/BlockMeta.java
package pw.switchcraft.plethora.integration.vanilla.meta.block;

import net.minecraft.block.Block;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;
import pw.switchcraft.plethora.api.meta.BasicMetaProvider;

import javax.annotation.Nonnull;
import java.util.HashMap;
import java.util.Map;

public final class BlockMeta extends BasicMetaProvider<Block> {
	public BlockMeta() {
		super("Provide the registry name, display name and translation key of a block.");
	}

	@Nonnull
	@Override
	public Map<String, ?> getMeta(@Nonnull Block block) {
		return getBasicMeta(block);
	}

	@Nonnull
	public static Map<String, ?> getBasicMeta(@Nonnull Block block) {
		HashMap<String, Object> data = new HashMap<>(3);

		Identifier name = Registry.BLOCK.getId(block);
		data.put("name", name.toString());

		data.put("displayName", block.getName().getString());
		data.put("translationKey", block.getTranslationKey());

		return data;
	}
}

File: src/main/java/io/sc3/plethora/integration/vanilla/meta/block/BlockMeta.kt
package io.sc3.plethora.integration.vanilla.meta.block

import io.sc3.plethora.api.meta.BasicMetaProvider
import net.minecraft.block.Block
import net.minecraft.registry.Registries

object BlockMeta : BasicMetaProvider<Block>(
  description = "Provide the registry name, display name and translation key of a block."
) {
  override fun getMeta(target: Block): Map<String, *> {
    val name = Registries.BLOCK.getId(target)
    return mapOf(
      "name"           to name.toString(),
      "displayName"    to target.name.string,
      "translationKey" to target.translationKey
    )
  }
}
